### PR TITLE
Avoid unnecessary call to EC2

### PR DESCRIPTION
**Pull request by mogren**
*Description of changes:*

No need to add 0 IPs.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/awsutils/awsutils.go [hunk 1]
```diff
@@ -866,6 +866,11 @@ func (cache *EC2InstanceMetadataCache) AllocIPAddresses(eniID string, numIPs int
 		needIPs = ipLimit
 	}
 
+	// If we don't need any more IPs, exit
```
Comment (tiffanyfay) [inline pkg/awsutils/awsutils.go]: 864 has no check for err != nil.

File: pkg/awsutils/awsutils.go [hunk 2]
```diff
@@ -866,6 +866,11 @@ func (cache *EC2InstanceMetadataCache) AllocIPAddresses(eniID string, numIPs int
 		needIPs = ipLimit
```
Comment (tiffanyfay) [inline pkg/awsutils/awsutils.go]: Should we log to let it be known that the ipLimit is less than the needed #?

File: pkg/awsutils/awsutils.go (hunk 1 above)
Comment (mogren) [reply pkg/awsutils/awsutils.go]: We should handle unknown instances better, but that's a different change.

File: pkg/awsutils/awsutils.go (hunk 2 above)
Comment (mogren) [reply pkg/awsutils/awsutils.go]: Right now we log that it the `tryAssignIPs()`

**Pull request merged**